packages/forms/src/widgets/MultiSelectTagWidget/MultiSelectTagWidget.js

```diff
@@ -218,132 +149,160 @@ class MultiSelectTagWidget extends React.Component {
 		);
 	}
 
-	scrollDropDownIfRequired(selectedIndex, direction) {
-		const dropDown = this.component.querySelector('.dropdown-menu');
-		const scrollPosition = dropDown.scrollTop;
-		const dropDownHeight = dropDown.clientHeight;
-		const itemStart = (selectedIndex * DROP_DOWN_ITEM_HEIGHT) + DROP_DOWN_PADDING;
-		const itemEnd = ((selectedIndex + 1) * DROP_DOWN_ITEM_HEIGHT) + DROP_DOWN_PADDING;
+	getDropdownItems(suggestions) {
+		let items = [];
+		if (this.state.withCategory) {
+			const itemsMap = {};
+			const groupBy = this.props.options.groupBy;
+
+			suggestions.forEach((item) => {
+				itemsMap[item[groupBy]] = itemsMap[item[groupBy]] || [];
+				itemsMap[item[groupBy]].push({ ...item, title: item.label });
+			});
+
+			Object.keys(itemsMap).forEach((category) => {
+				items.push({ title: category, suggestions: itemsMap[category] });
+			});
+		} else if (suggestions && suggestions.length > 0) {
+			items = suggestions.map(item => ({ ...item, title: item.label }));
+		} else {
+			items = suggestions;
+		}
+		return items;
+	}
+
+	updateSuggestions(value, props) {
+		const filterText = value === undefined ? this.state.filterText : value;
+		const currentProps = props || this.props;
+
+		let suggestions = filterOptions(currentProps);
+
+		if (filterText) { // filter with user's input
+			const escapedValue = escapeRegexCharacters(filterText.trim());
+			const regex = new RegExp(escapedValue, 'i');
+			suggestions = suggestions.filter(item => regex.test(item.label));
+			if (suggestions.length === 0 && currentProps.schema.createIfNoneMatch) {
+				suggestions.push({ label: `${filterText} (new)`, value: filterText });
+			}
+		}
+
+		this.setState({
```

Comment:
please us function setState signature since this state is derrived from a precedent state (you are using this.state.category)

https://reactjs.org/docs/react-component.html#setstate